### ISSUE TITLE
fluent-sqlserver

### ISSUE THREAD

**Issue by richgabrielli** [vapor/fluent#473]
Any thoughts / ambitions on creating connectivity to Microsoft SQL Server?

**Comment by tanner0101** [comment]
I would love to support this. Just not sure how to go about it. Ideally we would create a native protocol client using SwiftNIO so that we can take advantage of non-blocking IO. Or, if MSSQL already has a non-blocking C lib we could use that would be great as well.

**Comment by richgabrielli** [comment]
I found this, but I am not sure if it is what you are looking for: https://msdn.microsoft.com/en-us/library/cc425493(v=exchg.80).aspx

**Comment by richgabrielli** [comment]
This is more specific to SQL Server. First one may not be a good link: https://msdn.microsoft.com/en-us/library/ee210043(v=sql.105).aspx

**Comment by vzsg** [comment]
The only C driver library I'm aware of is [freetds](http://www.freetds.org/). I played around with it months ago and it was a massive pain to interact with from Swift, I gave up pretty soon.

The protocol it implements is documented here: [MS-TDS](https://msdn.microsoft.com/en-us/library/dd304523.aspx)

**Comment by richgabrielli** [comment]
Going with the assumption that there is no viable C driver, Are the document links in any way helpful to get started or we they not what you were looking for?

**Comment by Mordil** [comment]
After digging around MSFT's technical documents, I think I found the actual wire protocol definition: https://sqlprotocoldoc.blob.core.windows.net/productionsqlarchives/MS-SSTDS/[MS-SSTDS].pdf

Any work to support this is _probably_ going to go through the Swift Server Working Group before being implemented for Vapor.

**Comment by tanner0101** [comment]
Agreed. Linking the forum pitch here for reference: https://forums.swift.org/t/sql-server-driver/20327

**Comment by richgabrielli** [comment]
I am going to raise my hand again for SQL Server. It is one of the most widely used databases in the enterprise.

**Comment by richardpauljones** [comment]
I have made a start at - 
http://www.binaryrefinery.com

**Comment by Nash0x7E2** [comment]
Is there any updates to this? Would love to have MSSQL support in Vapor

**Comment by Mordil** [comment]
This has been pitched to the SSWG, and there's been some traffic on it.

Further tracking should be done via the Swift on Server Working Group forums at https://forums.swift.org/t/sql-server-driver/20327/.

Once a SQL Server driver has been authored, then Vapor will be able to write a Fluent driver to work with it.

**Comment by Nash0x7E2** [comment]
Noted, thank you. New to vapor so looking forward to contributing where possible :)
